config/builtin_locks.go

```diff
@@ -0,0 +1,26 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/gruntwork-io/terragrunt/locks"
+	"github.com/gruntwork-io/terragrunt/locks/dynamodb"
+)
+
+// lockFactory provides an implementation of Lock with the provided configuration map
+type lockFactory func(map[string]string) (locks.Lock, error)
```

Comment:
There wasn't a reason to export it, it's only used within this file and package.